service/dap/server.go

```diff
@@ -1646,7 +1662,16 @@ func (s *Server) convertVariableWithOpts(v *proc.Variable, qualifiedNameOrExpr s
 			variablesReference = maybeCreateVariableHandle(v)
 		}
 	case reflect.String:
-		// TODO(polina): implement auto-loading here
+		if evalContext == "repl" || evalContext == "variables" {
+			// TODO(hyangah): Copy Value from WATCH panel triggers an evaluate request in 'watch' mode,
+			// not in 'variables' mode. Either apply this in watch mode or after enabling the clipboard support,
+			// apply this in clipboard mode. When clipboard support is enabled, Copy Value from vscode VARIABLES
+			// panel also uses 'clipboard' mode. (Not sure where 'variables' mode will be used after then).
+			if strVal := constant.StringVal(v.Value); v.Len > int64(len(strVal)) {
+				value = reloadVariable(v, qualifiedNameOrExpr)
```

Comment:
Is this code path going to run on the variables returned by `s.debugger.FunctionArguments`? If it, is it will stop working really soon.

Reply:
Yes, I can imagine users would want to inspect function args further by querying from Debug Console.
What will be the new recommendation to achieve that in the future? 
Will the change affect other part of the reloadVariable too, or only be specific to string type args?

Reply:
It won't work for anything that is stored in a register, I'm not sure if there's any other code path that calls reloadVariable on a variable returned by `s.debugger.FunctionArguments`. It should use the `loadValue` method of `*proc.Variable` instead of calling EvalVariableInScope.

Reply:
This specific code path will work with the variable returned by `s.debugger.EvalVariableInScope` from https://github.com/go-delve/delve/blob/66782f9b27318b39130032f840cd684832ee05f8/service/dap/server.go#L1799. When evaluating a func arg name, does `EvalVariableInScope` do anything special to handle variable stored in a register?

`s.debugger.FunctionArguments` is used when handling the Scopes request (https://github.com/go-delve/delve/blob/66782f9b27318b39130032f840cd684832ee05f8/service/dap/server.go#L1314). It's possible that the user expands the args, editor sends Variables requests, `convertVariable` is called using the variable returned from the prior `FunctionArguments` call.  The string reload added in this PR will not be applied in this execution path.

TODO [here](https://github.com/go-delve/delve/blob/66782f9b27318b39130032f840cd684832ee05f8/service/dap/server.go#L1572 ) mentions that and it now sounds like a requirement for 1.17, not just an optimization.

Reply:
> When evaluating a func arg name, does EvalVariableInScope do anything special to handle variable stored in a register?

Yes, a lot, and the address of the variable isn't going to be a real address so it can't be passed back to EvalVariableInScope like reloadVariable does.

Reply:
> TODO here mentions that and it now sounds like a requirement for 1.17, not just an optimization.

My impression was that before this commit reloadVariable will never be called on variables returned by FunctionArgumens, only their children variables, which would be fine.

Reply:
Moved the logic out of this reloadVariable part to make it clear the code path added by this PR doesn't run as part of  general variable reloading.